File: java/src/sdk/dataobjects/paypal/payflow/BillTo.java
package paypal.payflow;

/**
 * Used for Billing Address information
 * <p>Billing address is Cardholder's address information.</p>
 * <p>Following example shows how to use BillTo.</p>
 *
 *  .................
 * // inv is the Invoice object.
 * .................
 *  * <p>
 * // Set the Billing Address details.
 * BillTo bill = new BillTo();
 * bill.setBillToStreet( "123 Main St.");
 * bill.setBillToZip("12345");
 * inv.setBillTo (bill);
 * .................
 */

public final class BillTo extends Address {

    private String billToHomePhone;
    private String billToCompanyName;

    /**
     * Constructor.
     * <p>Billing address is Cardholder's address information.</p>
     */
    public BillTo() {
    }

    /**
     * Gets the Billing Street.
     * <p>Cardholder's billing street address
     * (used for AVS and reporting).</P>
     *
     * @return String
     *  <p>Maps to Payflow Parameter: BILLTOSTREET</p>
     */
    public String getBillToStreet() {
        return super.getAddressStreet();
    }

    /**
     * Sets the Billing Street.
     * <p>Cardholder's billing street address
     * (used for AVS and reporting).</P>
     *
     * @param billToStreet String
     *  <p>Maps to Payflow Parameter: BILLTOSTREET</p>
     */
    public void setBillToStreet(String billToStreet) {
        super.setAddressStreet(billToStreet);
    }

    /**
     * Gets the Billing Street2.
     * <p>Cardholder's billing 2nd line street address.</P>
     *
     * @return String
     *  <p>Maps to Payflow Parameter: BILLTOSTREET2</p>
     */
    public String getBillToStreet2() {
        return super.getAddressStreet2();
    }

    /**
     * Sets the Billing Street.
     * <p>Cardholder's billing 2nd line street address.</P>
     *
     * @param billToStreet2 String
     *  <p>Maps to Payflow Parameter: BILLTOSTREET2</p>
     */
    public void setBillToStreet2(String billToStreet2) {
        super.setAddressStreet2(billToStreet2);
    }

    /**
     * Gets the city
     * <p>Cardholder's billing city.</P>
     *
     * @return String
     *  <p>Maps to Payflow Parameter: BILLTOCITY</p>
     */
    public String getBillToCity() {
        return super.getAddressCity();
    }

    /**
     * Sets the city
     * <p>Cardholder's billing city.</P>
     *
     * @param billToCity String
     *  <p>Maps to Payflow Parameter: BILLTOCITY</p>
     */
    public void setBillToCity(String billToCity) {
        super.setAddressCity(billToCity);
    }

    /**
     * Gets the State
     * <p>Cardholder's billing state code.</P>
     *
     * @return String
     *  <p>Maps to Payflow Parameter: BILLTOSTATE</p>
     */
    public String getBillToState() {
        return super.getAddressState();
    }

    /**
     * Sets the State
     * <p>Cardholder's billing state code.</p>
     *
     * @param billToState String
     *  <p>Maps to Payflow Parameter: BILLTOSTATE</p>
     */
    public void setBillToState(String billToState) {
        super.setAddressState(billToState);
    }

    /**
     * Gets the Billing Zip.
     * <p>Account holder's 5- to 9-digit postal code
     * (called ZIP code in the USA).
     * Do not use spaces, dashes,
     * or non-numeric characters.
     * The postal code is verified by the
     * AVS service.</P>
     *
     * @return String
     *  <p>Maps to Payflow Parameter: BILLTOZIP</p>
     */
    public String getBillToZip() {
        return super.getAddressZip();
    }

    /**
     * Sets the Billing Zip.
     * <p>UAccount holder's 5- to 9-digit postal code
     * (called ZIP code in the USA).
     * Do not use spaces, dashes,
     * or non-numeric characters.
     * The postal code is verified by the
     * AVS service.</P>
     *
     * @param billToZip String
     *  <p>Maps to Payflow Parameter: BILLTOZIP</p>
     */
    public void setBillToZip(String billToZip) {
        super.setAddressZip(billToZip);
    }

    /**
     * Gets the first name.
     * <p>Cardholder's first name.</P>
     *
     * @return firstName String
     *  <p>Maps to Payflow Parameter: BILLTOFIRSTNAME</p>
     */
    public String getBillToFirstName() {
        return super.getAddressFirstName();
    }

    /**
     * Sets the first name.
     * <p>Cardholder's first name.</P>
     *
     * @param billToFirstName String
     *  <p>Maps to Payflow Parameter: BILLTOFIRSTNAME</p>
     */
    public void setBillToFirstName(String billToFirstName) {
        super.setAddressFirstName(billToFirstName);
    }

    /**
     * Gets the middle name.
     * <p>Cardholder's middle name.</P>
     *
     * @return billToMiddleName String
     *  <p>Maps to Payflow Parameter: BILLTOMIDDLENAME</p>
     */
    public String getBillToMiddleName() {
        return super.getAddressMiddleName();
    }

    /**
     * Sets the middle name.
     * <p>Cardholder's middle name.</P>
     *
     * @param billToMiddleName String
     *  <p>Maps to Payflow Parameter: BILLTOMIDDLENAME</p>
     */
    public void setBillToMiddleName(String billToMiddleName) {
        super.setAddressMiddleName(billToMiddleName);
    }

    /**
     * Gets the last name.
     * <p>Cardholder's last name.</P>
     *
     * @return BillToLastName String
     *  <p>Maps to Payflow Parameter: BILLTOLASTNAME</p>
     */
    public String getBillToLastName() {
        return super.getAddressLastName();
    }

    /**
     * Sets the last name.
     * <p>Cardholder's last name.</P>
     *
     * @param billToLastName String
     *  <p>Maps to Payflow Parameter: BILLTOLASTNAME</p>
     */
    public void setBillToLastName(String billToLastName) {
        super.setAddressLastName(billToLastName);
    }

    /**
     * Gets the billing phone number.
     * <p>Cardholder's telephone number.</P>
     *
     * @return billToPhone String
     *  <p>Maps to Payflow Parameter: PHONENUM</p>
     */
    public String getBillToPhone() {
        return super.getAddressPhone();
    }

    /**
     * Sets the billing phone number.
     * <p>Cardholder's telephone number.</P>
     *
     * @param billToPhone String
     *  <p>Maps to Payflow Parameter: PHONENUM</p>
     */
    public void setBillToPhone(String billToPhone) {
        super.setAddressPhone(billToPhone);
    }

    /**
     * Gets the Billing Phone2.
     * <p>Cardholder's 2nd telephone number.</P>
     *
     * @return billToPhone2 String
     *  <p>Maps to Payflow Parameter: BILLTOPHONE2</p>
     */
    public String getBillToPhone2() {
        return super.getAddressPhone2();
    }

    /**
     * Sets the Billing Phone2.
     * <p>Cardholder's 2nd telephone number.</p>
     *
     * @param billToPhone2 String
     *  <p>Maps to Payflow Parameter: BILLTOPHONE2</p>
     */
    public void setBillToPhone2(String billToPhone2) {
        super.setAddressPhone2(billToPhone2);
    }

    /**
     * Gets the Billing Fax
     * <p>Cardholder's fax address.</P>
     *
     * @return fax String
     *  <p>Maps to Payflow Parameter: BILLTOFAX</p>
     */
    public String getBillToFax() {
        return super.getAddressFax();
    }

    /**
     * Sets the Billing Fax
     * <p>Cardholder's fax address.</P>
     *
     * @param billToFax String
     *  <p>Maps to Payflow Parameter: BILLTOFAX</p>
     */
    public void setBillToFax(String billToFax) {
        super.setAddressFax(billToFax);
    }

    /**
     * Gets the Billing Email.
     * <p>Cardholder's e-mail address</P>
     *
     * @return email String
     *  <p>Maps to Payflow Parameter: BILLTOEMAIL</p>
     */
    public String getBillToEmail() {
        return super.getAddressEmail();
    }

    /**
     * Sets the Billing Email.
     * <p>Cardholder's e-mail address</P>
     *
     * @param billToEmail String
     *  <p>Maps to Payflow Parameter: BILLTOEMAIL</p>
     */
    public void setBillToEmail(String billToEmail) {
        super.setAddressEmail(billToEmail);
    }

    /**
     * Gets the Billing Country.
     * <p>Cardholder's billing country code</P>
     *
     * @return billToCountry String
     *  <p>Maps to Payflow Parameter: BILLTOCOUNTRY</p>
     */
    public String getBillToCountry() {
        return super.getAddressCountry();
    }

    /**
     * Sets the Billing Country.
     * <p>Cardholder's billing country code</P>
     *
     * @param billToCountry String
     *  <p>Maps to Payflow Parameter: BILLTOCOUNTRY</p>
     */
    public void setBillToCountry(String billToCountry) {
        super.setAddressCountry(billToCountry);
    }

    /**
     * Gets the Billing HomePhone.
     * <p>Cardholder's home telephone number.</P>
     *
     * @return homePhone String
     *  <p>Maps to Payflow Parameter: HOMEPHONE</p>
     */
    public String getBillToHomePhone() {
        return billToHomePhone;
    }

    /**
     * Sets the Billing HomePhone.
     * <p>Cardholder's home telephone number.</P>
     *
     * @param billToHomePhone String
     *  <p>Maps to Payflow Parameter: HOMEPHONE</p>
     */
    public void setBillToHomePhone(String billToHomePhone) {
        this.billToHomePhone = billToHomePhone;
    }

    /**
     * Gets the Company Name.
     *
     * @return String
     *  <p>Maps to Payflow Parameter: COMPANYNAME</p>
     */

    public String getBillToCompanyName() {
        return billToCompanyName;
    }

    /**
     * Sets the Company Name.
     *
     * @param billToCompanyName String
     *  <p>Maps to Payflow Parameter: COMPANYNAME</p>
     */
    public void setBillToCompanyName(String billToCompanyName) {
        this.billToCompanyName = billToCompanyName;
    }


    /**
     * This method copies the common contents
     * from billing to shipping address.
     * <p>This method can be used to
     * populate the shipping addresses directly
     * from the billing addresses when
     * both are the same.</p>
     *
     * @return shipTo ShipTo
     *  ................
     * //bill is the object of
     * //billTo populated with
     * //the billing addresses.
     * ................
     *  * <p>
     * ShipTo ship;
     *  * <p>
     * //Populate shipping addresses
     * //from billing addresses.
     * ship = bill.copy();
     *  * <p>
     * ................
     */

    public ShipTo copy() {
        ShipTo copyObject = new ShipTo();
        copyObject.setAddressCity(this.getAddressCity());
        copyObject.setAddressCountry(this.getAddressCountry());
        copyObject.setAddressEmail(this.getAddressEmail());
        copyObject.setAddressFax(this.getAddressFax());
        copyObject.setAddressFirstName(this.getAddressFirstName());
        copyObject.setAddressLastName(this.getAddressLastName());
        copyObject.setAddressMiddleName(this.getAddressMiddleName());
        copyObject.setAddressPhone2(this.getAddressPhone2());
        copyObject.setAddressPhone(this.getAddressPhone());
        copyObject.setAddressState(this.getAddressState());
        copyObject.setAddressStreet(this.getAddressStreet());
        copyObject.setAddressStreet2(this.getAddressStreet2());
        copyObject.setAddressZip(this.getAddressZip());
        return copyObject;
    }

    protected void generateRequest() {
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_STREET, this.getBillToStreet()));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_STREET2, this.getBillToStreet2()));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_CITY, this.getBillToCity()));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_STATE, this.getBillToState()));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_BILLTOCOUNTRY, this.getBillToCountry()));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_ZIP, this.getBillToZip()));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_PHONENUM, this.getBillToPhone()));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_BILLTOPHONE2, this.getBillToPhone2()));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_EMAIL, this.getBillToEmail()));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_FAX, this.getBillToFax()));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_FIRSTNAME, this.getBillToFirstName()));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_MIDDLENAME, this.getBillToMiddleName()));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_LASTNAME, this.getBillToLastName()));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_HOMEPHONE, billToHomePhone));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_COMPANYNAME, billToCompanyName));
    }
}

File: dotNET/PayflowProSDK/PayPal/Payments/DataObjects/BillTo.cs
#region "Imports"

using System;
using PayPal.Payments.Common.Utility;
using PayPal.Payments.Common.Exceptions;

#endregion

namespace PayPal.Payments.DataObjects
{
	/// <summary>
	/// Used for Billing Address information
	/// </summary>
	/// <remarks>Billing address is Cardholder's address information.</remarks>
	/// <example>
	/// <para>Following example shows how to use BillTo.</para>
	/// <code lang="C#" escaped="false">
	///  .................
	///  //Inv is the Invoice object.
	///  .................
	///  
	///	//Set the Billing Address details.
	///	BillTo Bill = new BillTo();
	///	Bill.BillToStreet = "123 Main St.";
	///	Bill.BillToZip = "12345";
	///	Inv.BillTo = Bill;
	///	.................
	/// </code>
	/// <code lang="Visual Basic" escaped="false">
	///  .................
	///  'Inv is the Invoice object.
	///  .................
	///  
	///	'Set the Billing Address details.
	///	Dim Bill As BillTo = New BillTo
	///	Bill.BillToStreet = "123 Main St."
	///	Bill.BillToZip = "12345"
	///	Inv.BillTo = Bill
	///	.................
	/// </code>
	/// </example>
	public sealed class BillTo : Address
	{
		#region "Member variables"

		/// <summary>
		/// Home phone
		/// </summary>
		private String mBillToHomePhone;

		/// <summary>
		/// Company Name
		/// </summary>
		private String mBillToCompanyName;

		#endregion

		#region "Properties"

		/// <summary>
		/// Gets, Sets Billing City.
		/// </summary>
		/// <remarks>
		/// <para>Cardholder’s billing city.</para>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>BILLTOCITY</code>
		/// </remarks>
		public String BillToCity
		{
			get { return base.AddressCity; }
			set { base.AddressCity = value; }
		}

		/// <summary>
		/// Gets, Sets Billing Country.
		/// </summary>
		/// <remarks>
		/// <para>Cardholder’s billing country code</para>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>BILLTOCOUNTRY</code>
		/// </remarks>
		public String BillToCountry
		{
			get { return base.AddressCountry; }
			set { base.AddressCountry = value; }
		}

		/// <summary>
		/// Gets, Sets Billing Email.
		/// </summary>
		/// <remarks>
		/// <para>Cardholder’s e-mail address</para>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>BILLTOEMAIL</code>
		/// </remarks>
		public String BillToEmail
		{
			get { return base.AddressEmail; }
			set { base.AddressEmail = value; }
		}

		/// <summary>
		/// Gets, Sets Billing Fax
		/// </summary>
		/// <remarks>
		/// <para>Cardholder’s fax address.</para>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>BILLTOFAX</code>
		/// </remarks>
		public String BillToFax
		{
			get { return base.AddressFax; }
			set { base.AddressFax = value; }
		}

		/// <summary>
		/// Gets, Sets Billing Firstname.
		/// </summary>
		/// <remarks>
		/// <para>Cardholder’s first name.</para>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>BILLTOFIRSTNAME</code>
		/// </remarks>
		public String BillToFirstName
		{
			get { return base.AddressFirstName; }
			set { base.AddressFirstName = value; }
		}

		/// <summary>
		/// Gets, Sets Billing Lastname.
		/// </summary>
		/// <remarks>
		/// <para>Cardholder’s last name.</para>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>BillTOLASTNAME</code>
		/// </remarks>
		public String BillToLastName
		{
			get { return base.AddressLastName; }
			set { base.AddressLastName = value; }
		}

		/// <summary>
		/// Gets, Sets Billing Middlename.
		/// </summary>
		/// <remarks>
		/// <para>Cardholder’s middle name.</para>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>BILLTOMIDDLENAME</code>
		/// </remarks>
		public String BillToMiddleName
		{
			get { return base.AddressMiddleName; }
			set { base.AddressMiddleName = value; }
		}

		/// <summary>
		/// Gets, Sets Billing Phone2.
		/// </summary>
		/// <remarks>
		/// <para>Cardholder’s 2nd telephone number.</para>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>BILLTOPHONE2</code>
		/// </remarks>
		public String BillToPhone2
		{
			get { return base.AddressPhone2; }
			set { base.AddressPhone2 = value; }
		}

		/// <summary>
		/// Gets, Sets Billing Phonenum
		/// </summary>
		/// <remarks>
		/// <para>Cardholder’s telephone number.</para>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>BILLTOPHONENUM</code>
		/// </remarks>
		public String BillToPhone
		{
			get { return base.AddressPhone; }
			set { base.AddressPhone = value; }
		}

		/// <summary>
		/// Gets, Sets Billing State.
		/// </summary>
		/// <remarks>
		/// <para>Cardholder’s billing state code.</para>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>BILLTOSTATE</code>
		/// </remarks>
		public String BillToState
		{
			get { return base.AddressState; }
			set { base.AddressState = value; }
		}

		/// <summary>
		/// Gets, Sets Billing Street.
		/// </summary>
		/// <remarks>
		/// <para>Cardholder’s billing street address 
		/// (used for AVS and reporting).</para>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>BILLTOSTREET</code>
		/// </remarks>
		public String BillToStreet
		{
			get { return base.AddressStreet; }
			set { base.AddressStreet = value; }
		}

		/// <summary>
		/// Gets, Sets Billing Street2
		/// </summary>
		/// <remarks>
		/// <para>Cardholder’s billing street 2 address 
		/// (used for AVS and reporting).</para>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>BILLTOSTREET2</code>
		/// </remarks>
		public String BillToStreet2
		{
			get { return base.AddressStreet2; }
			set { base.AddressStreet2 = value; }
		}

		/// <summary>
		/// Gets, Sets Billing Zip.
		/// </summary>
		/// <remarks>
		/// <para>Account holder’s 5- to 9-digit postal code
		///  (called ZIP code in the USA). 
		///  Do not use spaces, dashes, 
		///  or non-numeric characters.
		///  The postal code is verified by the 
		///  AVS service.</para>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>BILLTOZIP</code>
		/// </remarks>
		public String BillToZip
		{
			get { return base.AddressZip; }
			set { base.AddressZip = value; }
		}

		/// <summary>
		/// Gets, Sets Billing HomePhone.
		/// </summary>
		/// <remarks>
		/// <para>Cardholder’s home telephone number.</para>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>HOMEPHONE</code>
		/// </remarks>
		public String BillToHomePhone
		{
			get { return mBillToHomePhone; }
			set { mBillToHomePhone = value; }
		}

		/// <summary>
		/// Gets, Sets  CompanyName.
		/// </summary>
		/// <remarks>
		/// <para>Company Name.</para>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>COMPANYNAME</code>
		/// </remarks>
		public String BillToCompanyName
		{
			get { return mBillToCompanyName; }
			set { mBillToCompanyName = value; }
		}

		#endregion

		#region "Constructors"

		/// <summary>
		/// Constructor.
		/// </summary>
		/// <remarks>Billing address is Cardholder's address information.</remarks>
		/// <example>
		/// <para>Following example shows how to use BillTo.</para>
		/// <code lang="C#" escaped="false">
		///  .................
		///  //Inv is the Invoice object.
		///  .................
		///  
		///	//Set the Billing Address details.
		///	BillTo Bill = new BillTo();
		///	Bill.BillToStreet = "123 Main St.";
		///	Bill.BillToZip = "12345";
		///	Inv.BillTo = Bill;
		///	.................
		/// </code>
		/// <code lang="Visual Basic" escaped="false">
		///  .................
		///  'Inv is the Invoice object.
		///  .................
		///  
		///	'Set the Billing Address details.
		///	Dim Bill As BillTo = New BillTo
		///	Bill.BillToStreet = "123 Main St."
		///	Bill.BillToZip = "12345"
		///	Inv.BillTo = Bill
		///	.................
		/// </code>
		/// </example>
		public BillTo()
		{
		}

		#endregion

		#region "Utility function"

		/// <summary>
		/// This method copies the common contents
		/// from billing to shipping address.
		/// </summary>
		/// <returns>ShipTo object</returns>
		/// <remarks>This method can be used to 
		/// populate the shipping addresses directly 
		/// from the billing addresses when 
		/// both are the same.</remarks>
		/// <example>
		/// <code lang="C#" escaped="false">
		/// 
		///		................
		///		//Bill is the object of
		///		//BillTo populated with 
		///		//the billing addresses.
		///		................
		///		
		///		
		///		ShipTo Ship;
		///		
		///		//Populate shipping addresses
		///		//from billing addresses.
		///		Ship = Bill.Copy();
		///		
		///		................
		/// 
		/// </code>
		/// <code lang="Visual Basic" escaped="false">
		/// 
		///		................
		///		'Bill is the object of
		///		'BillTo populated with 
		///		'the billing addresses.
		///		................
		///		
		///		
		///		Dim Ship As ShipTo
		///		
		///		'Populate shipping addresses
		///		'from billing addresses.
		///		Ship = Bill.Copy()
		///		
		///		................
		/// 
		/// </code>
		/// </example>
		/// <seealso cref="ShipTo"/>
		public ShipTo Copy()
		{
			ShipTo CopyObject = new ShipTo();
			CopyObject.AddressCity = AddressCity;
			CopyObject.AddressCountry = AddressCountry;
			CopyObject.AddressEmail = AddressEmail;
			CopyObject.AddressFax = AddressFax;
			CopyObject.AddressFirstName = AddressFirstName;
			CopyObject.AddressLastName = AddressLastName;
			CopyObject.AddressMiddleName = AddressMiddleName;
			CopyObject.AddressPhone2 = AddressPhone2;
			CopyObject.AddressPhone = AddressPhone;
			CopyObject.AddressState = AddressState;
			CopyObject.AddressStreet = AddressStreet;
			CopyObject.AddressStreet2 = AddressStreet2;
			CopyObject.AddressZip = AddressZip;
			return CopyObject;
		}

		#endregion

		#region "Core functions"

		/// <summary>
		/// Generates the transaction request.
		/// </summary>
		internal override void GenerateRequest()
		{
			try
			{
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_STREET, this.BillToStreet));
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_BILLTOSTREET2, this.BillToStreet2));
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_CITY, this.BillToCity));
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_STATE, this.BillToState));
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_BILLTOCOUNTRY, this.BillToCountry));
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_ZIP, this.BillToZip));
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_PHONENUM, this.BillToPhone));
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_BILLTOPHONE2, this.BillToPhone2));
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_EMAIL, this.BillToEmail));
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_FAX, this.BillToFax));
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_FIRSTNAME, this.BillToFirstName));
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_MIDDLENAME, this.BillToMiddleName));
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_LASTNAME, this.BillToLastName));
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_HOMEPHONE, mBillToHomePhone));
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_COMPANYNAME, mBillToCompanyName));
			}
			catch (BaseException)
			{
				throw;
			}
			catch (Exception Ex)
			{
				DataObjectException DEx = new DataObjectException(Ex);
				throw DEx;
			}
		}

		#endregion
	}
}